File: client/src/vector2d.ts
export class Vector2D {
    x: number;
    y: number;

    constructor(x: number, y: number) {
        this.x = x;
        this.y = y;
    }

    public add(otherVec: Vector2D) {
        this.x += otherVec.x;
        this.y += otherVec.y;
    }

    public subtract(otherVec: Vector2D) {
        this.x -= otherVec.x;
        this.y -= otherVec.y;
    }

    public equals(otherVec: Vector2D): boolean {
        return (this.x == otherVec.x && this.y == otherVec.y);
    }
}

File: vector2D.go
package main

type Vector2D struct {
	X float64 `msgpack:"x"`
	Y float64 `msgpack:"y"`
}

func (vec Vector2D) add(otherVec Vector2D) {
	vec.X += otherVec.X
	vec.Y += otherVec.Y
}

func (vec Vector2D) subtract(otherVec Vector2D) {
	vec.X -= otherVec.X
	vec.Y -= otherVec.Y
}

func (vec Vector2D) equals(otherVec Vector2D) bool {
	return (vec.X == otherVec.X && vec.Y == otherVec.Y)
}
